### ISSUE TITLE
Phylogenetic Tree

### ISSUE THREAD

**Issue by nsamhada** [JosephCrispell/homoplasyFinder#20]
Hello, it seems as though the phylogenetic trees I try to make are always faulty. I tried two different methods, and one provided this error:
Error in rJava::.jcall(javaHomoplasyFinderClass, method = "runHomoplasyFinderFromR",  : 
  java.lang.NumberFormatException: For input string: "1e-08)"

and the other provided this:
complete genome" isn't present as a tip label in the newick tree file.
        The sequence and tip IDs must match exactly.

What is a good way to build my phylogenetic tree using my aligned fasta sequences so that it can work as an input for the tool?

**Comment by JosephCrispell** [comment]
Hi,

Thanks for using `homoplasyFinder` and sorry it isn't working. `HomoplasyFinder` uses a standard [newick tree format](http://wiki.christophchamp.com/index.php?title=Newick_phylogenetic_tree_format#:~:text=The%20Newick%20phylogenetic%20tree%20format,famous%20English%20mathematician%20Arthur%20Cayley.).

Your first error is caused by your `tree` file have an extra bracket somewhere. The other error is caused by a tip label not being found in the `FASTA` file.

Most tree building software will be able to output a phylogeny in a newick format, here a few you could try [phylip](https://evolution.genetics.washington.edu/phylip.html), [RAxML](https://cme.h-its.org/exelixis/web/software/raxml/), and [ape](https://cran.r-project.org/web/packages/ape/ape.pdf). You could also load your tree into a software like [figtree](http://tree.bio.ed.ac.uk/software/figtree/), [icytree](https://icytree.org/), or [ape](https://cran.r-project.org/web/packages/ape/ape.pdf) and save it as a newick formatted tree file.

I hope the above is useful.

**Comment by nsamhada** [comment]
Thank you for you for your quick reply! 

Once I do convert the tree into Newick format, I get the error: "ERROR!! The following sequence name: "NC_045512.2 Severe acute respiratory syndrome coronavirus 2 isolate Wuhan-Hu-1, complete genome" isn't present as a tip label in the newick tree file.
        The sequence and tip IDs must match exactly.
Error: no more error handlers available (recursive errors?); invoking 'abort' restart"

The sequence name that is not found is for the reference genome. Do you think that has to do with it?

**Comment by JosephCrispell** [comment]
Hi,

That error means the sequence ID (`"NC_045512.2 Severe acute respiratory syndrome coronavirus 2 isolate Wuhan-Hu-1, complete genome"`) present in the `FASTA` file is not present in the `tree` file as a tip label. To check this you can open the `tree` file and search for it.

The quickest solution is just to open the FASTA file, remove the sequence and label for `"NC_045512.2 Severe acute respiratory syndrome coronavirus 2 isolate Wuhan-Hu-1, complete genome"` and save the file as a different `FASTA` file that you can then send to `homoplasyFinder`.

**Comment by nsamhada** [comment]
I tried to edit my fasta file to be formatted exactly how the newick tree file is formatted. 
For perspective, my tree file looks like this:
Tip labels:
  ERR4238443, SRR11828430, SRR11728625, SRR11494589, ERR4085015_ARTIC_medaka, ERR4238517, ...
Node labels:
  , 3, 1, 0, 0, 0, ...

And the error I'm getting now is: ERROR!! Unrecognised character state (_) at position: 49. 


My initial problem was that the tip labels were not exactly the same because in the fasta file the sequences were written as
"ERR4085015/ARTIC/medaka" and I edited the files to have them written the exact way. The character is somehow not recognized.
